samples/angular-todo/src/app/features/journey/webAuthn/webAuthn.component.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * angular-todo-prototype
+ *
+ * webAuthn.component.ts
+ *
+ * Copyright (c) 2021 ForgeRock. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+import { Component, Input, OnInit } from '@angular/core';
+import {
+  FRAuth,
+  FRLoginFailure,
+  FRLoginSuccess,
+  FRStep,
+  TokenManager,
+  UserManager,
+} from '@forgerock/javascript-sdk';
+import FRWebAuthn from 'packages/javascript-sdk/src/fr-webauthn';
+import { WebAuthnStepType } from 'packages/javascript-sdk/src/fr-webauthn/enums';
+import { UserService } from '../../../services/user.service';
+import { Router } from '@angular/router';
+
+/**
+ * Used to handle webAuthn journey
+ */
+@Component({
+  selector: 'app-webAuthn',
+  templateUrl: './webAuthn.component.html',
+})
+export class WebAuthnComponent implements OnInit {
+  @Input() webAuthnType: WebAuthnStepType;
+  @Input() step: FRStep;
+  isWebAuthnRegistration: boolean;
+  message: string;
+  header: string;
+  success: FRLoginSuccess;
+  failure: FRLoginFailure;
+
+  constructor(private router: Router, public userService: UserService) {}
+
+  async ngOnInit(): Promise<void> {
+    try {
+      switch (this.webAuthnType) {
+        case WebAuthnStepType.Registration:
+          this.isWebAuthnRegistration = true;
+          this.message = 'Your device will be used to verify your identity';
+          this.header = 'Registering your device';
+          this.step = await FRWebAuthn.register(this.step);
+          break;
+
+        case WebAuthnStepType.Authentication:
+          this.isWebAuthnRegistration = false;
+          this.message = 'Use your device to verify your identity';
+          this.header = 'Verifying your identity';
+          this.step = await FRWebAuthn.authenticate(this.step);
+          break;
```

Comment:
We should probably have the SDK INTEGRATION POINT JS Doc comment for this block of code, yeah?

Reply:
Regarding the first quesiton, it's done i'd misunderstood the question. Sorry, dont get the last comment

Reply:
Sorry, let me restate my comment for better clarity. Being that we are try to use JSDoc style comments through this code base to help provide hints to the developer using this to learn how to use the SDK, I think it would be good to have a comment here.

The comment could explain what may not be immediately obvious, like the try catch, the fact that the step is mutated, even when an error is thrown. Let's try to help the developer with comments to ensure they don't get stuck on the same thing you got tripped up on. Make sense?

These comments look like this:

```js
/** *********************************************************************
 * SDK INTEGRATION POINT
 * Summary: Short description
 * ----------------------------------------------------------------------
 * Details: Longer explanation
 ************************************************************************* */
```

Does that make sense? We try to liberally use the comments throughout the sample app codebase.

Reply:
ah got it,  not a problem i will add a comment explaning what the code does.

Reply:
done